sled-agent/src/dump_setup.rs

```diff
@@ -1129,13 +1134,49 @@ impl DumpSetupWorker {
         for path in
             glob::glob(debug_dir.as_ref().join("**/*").as_str())?.flatten()
         {
-            let meta = tokio::fs::metadata(&path).await?;
+            let meta = match tokio::fs::metadata(&path).await {
+                Ok(meta) => meta,
+                // "Not found" errors could be caused by TOCTTOU -- observing a
+                // file which gets renamed, removed, etc. We'll ignore these
+                // files.
+                Err(err)
+                    if matches!(err.kind(), std::io::ErrorKind::NotFound) =>
+                {
+                    continue;
+                }
+                // Other errors get propagated.
+                Err(err) => return Err(err.into()),
+            };
+
+            if !meta.is_file() {
+                // Ignore non-files (e.g. directories, symlinks)
+                continue;
+            }
```

Comment:
We previously *tried* to delete directories with `tokio::fs::remove_file`. This would cause the cleanup process to bail.

(If we end up with a "really old" directory in `crypt/debug`, this would prevent it from ever being cleanable again)